labs/b8.md

```diff
@@ -0,0 +1,84 @@
+>GnuPG allows you to encrypt and sign your data and communications; it features a versatile key management system, along with access modules for all kinds of public key directories.
+
+## Encryption and Decryption
+Encryption takes in input and a key, returning garbled output. Decryption takes in a key and the garbled output from encryption, recovering and returning the original input only if the decryption key is valid.
+
+![alt](https://docs.oracle.com/cd/E19424-01/820-4811/images/scrypt.gif)
+
+### Symmetric Cryptography
+In symmetric cryptography, the keys used for encryption and decryption are the same. 
+
+To try it out: 
+1. `gpg -c [FILE]` on any file to output a `[FILE].gpg` file which is the encrypted version of the inputted file. You'll need to enter a password when encrypting the file.
+2. `gpg -d [FILE].gpg` on the encrypted version of original file, upon which you'll need to enter the original password.
+
+In this GPG implementation, encryption and decryption of your file both require knowledge of a single password, which in this case serves as the symmetric key. 
+
+### Asymmetric Cryptography
+In asymmetric cryptography, two separate keys are respectively used for encryption and decryption. Typically, the two keys come in a public-private pair. The public key is made known publicly and used to encrypt data. Whereas, the private key is kept secret by the owner and used to decrypt data. Encryption of a file with a public key implies that only someone with the corresponding private key can the decrypt the resulting encrypted file.
+
+#### GPG Keyring Abstraction
+GPG uses a "keyring" as a centralized location to hold all of a user's keys. You'll need to add/import a key to your keyring if you wish to use it and keep it around. Similarly, if you wish to share a key with someone else, you can export your key (which makes a copy of your key) and have them import it to their keyring. 
+
+To try it out:
+1. `gpg --full-generate-key` to generate a GPG public-private key pair. It'll ask for a password.
+2. `gpg -r [RECIPIENT] -e [FILE]` which'll encrypt `[FILE]` with `[RECIPIENT]`'s public key (for now, try encrypting a file with your own public key). 
+3. `gpg -d [FILE].gpg` will search through your keyring and decrypt the file with the appropriate private key (if you possess the correct private key, of course). 
+*You don't need to specify which key to decrypt a file with because GPG-encrypted files and keys contain metadata that allow GPG to pick the correct key from the keyring to decrypt the file with.*
+
+### Signatures
+The asymmetric scheme involving encryption with public key and decryption with private key can also be reversed to implement signatures, which allows for the authenticity of some data to be verified. In this reversed scheme, the private key is used to sign a file, producing a signature on that file. And the corresponding public key is used to verify the signature. 
```

Comment:
s/some data/data/

you should probably also explain what a signature is, along with stating what it does